tests/RobotFramework/tests/cumulocity/log/workflow/log_upload.toml

```diff
@@ -20,6 +20,10 @@ on_error = "failed"
 
 [executing]
   action = "builtin"
+  on_success = "postprocess"      # on_success & on_error can be customized for builtin actions
+
+[postprocess]
+  script = "/usr/bin/log_upload.sh ${.payload.status}"
```

Comment:
I don't fully follow you.

- `action = "builtin"` is not deprecated yet: this is still working, but discouraged in the documentation.
- `operation = "builtin:<operation_name>` is really nicer from a user perspective. This is a personal point of view, but with some evidence: the documentation has been far simpler to write and there is less traps in using it.
- We are now in a position to make implicit the `action = "await-operation-completion"`, removing some burden to the user.
- I see smaller-grain actions like `download`, `file-copy` ... as a nice idea we really have to work on. But, I don't see any those as full-fledged operations (with an MQTT topic, capabilities, retain messages, states ...). Hence, no risk of confusion.

Reply:
> From the external user's perspective, the second one with just 2 states looks much simpler, as the additional `await_operation` state in the first approach is just boilerplate which adds no value to the user (unless I'm missing something here). 

Yes, you are missing something. __3__ steps are already required in the second case. Indeed, the `scheduled` step must be invoked as well as the executing step.

> The only additional thing that they get by using `operation` is the `input` and `input_script` support, which is also less useful here IMO, as there is little scope for input manipulation when the main operation and the builtin operation are of the same type (since the same input can be passed as-is).

This was the plan when starting this PR, but has been abandoned meantime. As a `builtin:<operation>` can only apply to the current operation there is no point to tune the input.

> Or is there some additional benefit with the `operation` approach that I'm missing?

Using `action = "builtin"` is really error prone unless you stick to a strict template.